src/_pytest/config/__init__.py

```diff
@@ -608,20 +610,33 @@ def _iter_rewritable_modules(package_files):
 
 
 class Config:
-    """ access to configuration values, pluginmanager and plugin hooks.  """
+    """
+    Access to configuration values, pluginmanager and plugin hooks.
+
+    :ivar PytestPluginManager pluginmanager: the plugin manager handles plugin registration and hook invocation.
+
+    :ivar argparse.Namespace option: access to command line option as attributes.
+
+    :ivar invocation_args: list of command-line arguments as passed to pytest.main()
+
+    :ivar invocation_plugins: list of extra plugins passed to pytest.main(), might be None
+
+    :ivar py.path.local invocation_dir: directory where pytest.main() was invoked from
+    """
 
     def __init__(self, pluginmanager):
-        #: access to command line option as attributes.
-        #: (deprecated), use :py:func:`getoption() <_pytest.config.Config.getoption>` instead
-        self.option = argparse.Namespace()
         from .argparsing import Parser, FILE_OR_DIR
 
+        self.option = argparse.Namespace()
+        self.invocation_args = None
```

Comment:
those should be passable as arguments, same for invocation dir,

i do wonder if a "invocation" should be a namedtuple in some way

Reply:
> those should be passable as arguments, same for invocation dir,

Sounds good, but then should we handle `plugins` here as well?

https://github.com/pytest-dev/pytest/blob/3e669a262a880acdc8a69cad0761e4f1925abe21/src/_pytest/config/__init__.py#L191-L196

Otherwise it is a bit strange to require a `plugins` and do nothing with it, just store it as an attribute.

> i do wonder if a "invocation" should be a namedtuple in some way

Sure, we can make a frozen `attr.s` class with that. So:

```python
@attr.s(frozen=True)
class InvocationParams:
    args = attr.ib()  # list of str
    dir = attr.ib()  # Path object
    plugins = attr.ib()  # list of str or None

class Config:
    def __init__(self, pluginmanager, args=None, plugins=None, dir=None):
        ...
```

Reply:
if we factor a invocation argument indicator into a own object, its components shouldnt be part of the config constructor

also we cant do anything with plugins at construction time as its likely incorrect and has a broken order(thanks to the config initialization phases)

Reply:
Done, let me know what you think